nvflare/private/fed/server/server_engine.py

```diff
@@ -130,45 +138,46 @@ def get_run_info(self) -> RunInfo:
         else:
             return None
 
-    def set_run_number(self, num):
-        # status = self.server.status
-        status = self.engine_info.status
-        if status == MachineStatus.STARTING or status == MachineStatus.STARTED:
-            return "run_number can not be changed during the FL app running."
+    def create_parent_connection(self, port):
+        while not self.parent_conn:
+            try:
+                address = ("localhost", port)
+                self.parent_conn = CommandClient(address, authkey="parent process secret password".encode())
+            except Exception as e:
+                time.sleep(1.0)
+                pass
 
-        run_folder = os.path.join(self.args.workspace, "run_" + str(num))
-        if os.path.exists(run_folder):
-            self.run_number = num
-            return "run number already exists. Set the FL run number to {}.".format(num)
-        else:
-            self.run_number = num
-            os.makedirs(run_folder)
-            return "Created a new run folder: run_{}".format(num)
+        threading.Thread(target=self.heartbeat_to_parent, args=[]).start()
```

Comment:
I see that you don't join this kind of threads. Is it safe? Could it hang?

Reply:
heartbeat_to_parent will stop. then it will be ok,